diff -up openssl-1.0.2j/ssl/ssl.h.nokrb5-abi openssl-1.0.2j/ssl/ssl.h
--- openssl-1.0.2j/ssl/ssl.h.nokrb5-abi	2016-10-07 11:33:36.000000000 +0200
+++ openssl-1.0.2j/ssl/ssl.h	2016-10-14 13:26:29.767624676 +0200
@@ -521,6 +521,9 @@ struct ssl_session_st {
 #  ifndef OPENSSL_NO_KRB5
     unsigned int krb5_client_princ_len;
     unsigned char krb5_client_princ[SSL_MAX_KRB5_PRINCIPAL_LENGTH];
+#  else
+    unsigned int unused_krb5_client_princ_len;
+    unsigned char unused_krb5_client_princ[SSL_MAX_KRB5_PRINCIPAL_LENGTH];
 #  endif                        /* OPENSSL_NO_KRB5 */
 #  ifndef OPENSSL_NO_PSK
     char *psk_identity_hint;
@@ -1521,6 +1524,8 @@ struct ssl_st {
 #  ifndef OPENSSL_NO_KRB5
     /* Kerberos 5 context */
     KSSL_CTX *kssl_ctx;
+#  else
+    void *unused_kssl_ctx;
 #  endif                        /* OPENSSL_NO_KRB5 */
 #  ifndef OPENSSL_NO_PSK
     unsigned int (*psk_client_callback) (SSL *ssl, const char *hint,
